app/services/sso/jwt.rb

```diff
@@ -37,7 +37,7 @@ def bearer_token_set?
     end
 
     def no_issuer?
-      !jwt_token.decoded_payload.key?('iss')
+      jwt_token.decoded_payload.present? && !jwt_token.decoded_payload.try(:key?, 'iss')
```

Comment:
again, no need for both `present?` and `try`

Reply:
I think we would require both, here we need to check if the token is present, if it is present then it must not have the `iss` key, therefore both are used.

Reply:
can the payload be anything other then `nil` or a hash? otherwise, this is equivalent:
```suggestion
      jwt_token.decoded_payload.present? && !jwt_token.decoded_payload.key?('iss')
```
because if the token isn't present the second part of the condition won't be checked at all, and if it is that means it isn't nil and we don't need to `try`.

Reply:
Makes sense! thanks.

Reply:
Maybe if it's not a hash, it should be rejected in decoding?

Reply:
Looking at the previous conversation on this thread https://github.com/theforeman/foreman/pull/7335#issuecomment-572055977 i think it is best to use the `try` method because we are not 100% sure if the input will always be a hash or nil. @tbrisker what you think?

Reply:
Giving that a closer look, I think, the values can either be `nil` or the `hash`. It can not be anything else because in the `decoded_payload` if there is any error we rescue it and set a nil value. Therefore I dont think we would require a `try` mothod. We can just do as @tbrisker has suggested. @ekohl would you agree?

Reply:
looks like payload doesn't have to be hash, especially if it's user generated. e.g.:

```
[51] pry(main)> token = JWT.encode('blah', nil, 'none')
=> "eyJhbGciOiJub25lIn0.ImFhc2Ei."
[52] pry(main)> JWT.decode(token, nil, false).first
=> "blah"
```
Perhaps instead of testing for `payload.present?` or `payload.nil?` we should test for `payload.is_a?(Hash)`? Or like @ekohl suggested, reject it when decoding (would that also be the case for non-OIDC JWT tokens or are those valid if the payload is not a hash)?

Reply:
I think we can validate in the `decode_token` method. JWT tokens must decode in the form of a Hash.

I have made the code changes, let me know what you folks think :)